Automattic/woocommerce-payments
# Hook into the Best Sales Day Notification to convert order totals

Fixes #2566 

#### Changes proposed in this Pull Request









- Update to add a filter which hooks into the `WC_Order_Query`, but only affects the output when it is being called in the context of the `NewSalesRecord` [inbox notification](https://woocommerce.github.io/code-reference/files/woocommerce-packages-woocommerce-admin-src-notes-newsalesrecord.html#source-view.15).

#### Testing instructions





* Install WP Crontrol on your site: https://wordpress.org/plugins/wp-crontrol/
* Make sure you are running using a host (e.g. Jurassic Tube), as running crons doesn't appear to work when the host is `localhost`.
* Note that the cron job checks orders from *yesterday*. If you have no multi-currency orders from yesterday, you could change [this line](https://github.com/woocommerce/woocommerce-admin/blob/38fbb4aedd0f21cf21ddaa6af93e48564fd35324/src/Notes/NewSalesRecord.php#L71) in the WC Admin code to 'today' to test with and then make a new order.
* Add some logging using `error_log()` to the `convert_order_prices` function to verify it is getting run, and to check it is converting order prices (for example, log the value of `$order->get_total()` before and after the call to `$order->set_total( number_format( $order->get_total() * ( 1 / $exchange_rate ), wc_get_price_decimals() ) );` on line 314 of `Compatibility.php`.
* Go to the `Tools -> Cron Events` section and look for the `wc_admin_daily` event. Click "Run Now".
* Check your logs to verify the function was hit, and it converted the total for any multi-currency orders to the store default currency.

-------------------
